src/Support/ColumnMetadata.php

```diff
@@ -0,0 +1,110 @@
+use Illuminate\Support\Collection;
+use Illuminate\Support\Str;
+use ReflectionClass;
+use ReflectionMethod;
+
+class ColumnMetadata
+{
+    public function __construct(
+        protected Model $baseModel,
+        /** @var RelationMetadata[] */
+        protected array $relations,
+        protected string $column,
+    ) {
+    }
+
+    public static function fromString(EloquentBuilder|Relation $subject, string $column): self
+    {
+
+        $parts = Str::of($column)->explode('.');
+
+        if ($parts->count() === 1) {
+            // --> No nested information
+            return new self($subject->getModel(), [], $column);
+        }
+
+        $relations = [];
+        $model = $subject->getModel();
+
+        foreach ($parts as $index => $part) {
+
+            $relationName = Str::camel($part);
+            $modelRelations = self::getRelations($model::class);
+
+            if ($modelRelations->contains($relationName)) {
+                $relation = $model->$relationName();
+                $model = $relation->getModel();
+
+                $relations[] = new RelationMetadata($relationName, $model);
+            } else {
+                // --> End of relation (further dots must be json nesting)
+                $remaining = $parts->skip($index + 1)->implode('.');
+                if (! empty($remaining)) {
+                    $remaining = '.'.$remaining;
+                }
+                $column = $part.$remaining;
+                break;
+            }
+        }
+
+        return new self($subject->getModel(), $relations, $column);
+
+    }
+
+    protected static function getRelations(string $modelClass): Collection
+    {
+        return collect((new ReflectionClass($modelClass))->getMethods(ReflectionMethod::IS_PUBLIC))
+            ->filter(function (ReflectionMethod $reflectionMethod) {
+                $returnType = (string) $reflectionMethod->getReturnType();
+
+                return $returnType != null && is_subclass_of($returnType, Relation::class);
+            })
+            ->map(fn (ReflectionMethod $reflectionMethod) => $reflectionMethod->getName());
+    }
+
+    public function hasRelations(): bool
+    {
+        return ! empty($this->relations);
+    }
+
+    public function getDottedRelation(): string
+    {
+        return collect($this->relations)
+            ->implode('name', '.');
+    }
+
+    public function getColumn(): string
+    {
+        return $this->column;
+    }
+
+    public function isJsonColumn(): bool
+    {
+        $model = collect($this->relations)->last()?->model ?? $this->baseModel;
+        $colum = Str::before($this->column, '.');
+
+        return str_contains($this->column, '.') || $model->hasCast($colum, [
+            'array',
+            'json',
+            'object',
+            'collection',
+            'encrypted:array',
+            'encrypted:collection',
+            'encrypted:json',
+            'encrypted:object',
```

Comment:
I think these would be good to have as a config.
And we should check if this can not also (additionally) be a JSON cast, if someone uses non-string casts. 
It could be that we need to include checks for those too, at least the built-in cast classes.
For extra fancyness, we could also allow the user to provide a custom callback on a facade, e.g. as you would customize functions in Fortify, to allow for extra flexibility.

Reply:
This would be useful, especially for custom casts.